### PR TITLE
Fix: data-highlight for select

### PR DESCRIPTION
**Pull request by saturnonearth**
Add highlight on hover for `Select` component, and switch over to using `data-[highlighted]` instead of `focus:`.

Issue that still remains:

CleanShot 2023-10-25 at 12 42 38@2x

`cursor: 'pointer'` being injected somewhere on the `Select` items, overriding the `cursor-default` class that is already present.

### CODE REVIEW

**Comment by huntabyte** [comment]
Changesets are only needed when changes are made to the cli @saturnonearth 😃

File: apps/www/src/lib/registry/default/ui/select/select-item.svelte
Comment (huntabyte) [inline apps/www/src/lib/registry/default/ui/select/select-item.svelte:21]: We shouldn't need `hover` classes here.

You also forgot to remove `focus` from one of the highlighted selectors 😃
should be - `data-[highlighted]:text-accent-foreground`

File: apps/www/src/lib/registry/new-york/ui/select/select-item.svelte
Comment (huntabyte) [inline apps/www/src/lib/registry/new-york/ui/select/select-item.svelte:21]: Same here, I don't think we need hover classes as highlighted should handle hover events and it can be weird if you leave your mouse on one item and then use your keyboard to navigate down, it will technically show two items as highlighted.

**Review by huntabyte — CHANGES_REQUESTED** [review]
Just a few minor things

File: apps/www/src/lib/registry/new-york/ui/select/select-item.svelte
Comment (saturnonearth) [reply apps/www/src/lib/registry/new-york/ui/select/select-item.svelte:21]: > Same here, I don't think we need hover classes as highlighted should handle hover event

Yea that makes sense, but without hover mouseover it does not highlight on hover - but I am not sure if that is because melt-ui needs to be updated or something else?

>  It can be weird if you leave your mouse on one item and then use your keyboard to navigate down, it will technically show two items as highlighted.

I agree, assuming highlighted works in both cases, that's fine. But if we _do_ need to use `hover:` I don't think it would be a big deal as  I _assume_ (possibly incorrectly) someone using a keyboard to navigate would be less likely to also be using the mouse for navigation.

**Comment by huntabyte** [comment]
This issue/PR is blocking this from working as we'd expect: https://github.com/melt-ui/melt-ui/issues/667 hopefully gets merged and released soon and I'll merge this in once we confirm it works as expected!

**Review by huntabyte — APPROVED** [review]

**Comment by saturnonearth** [comment]
omg can't believe I forgot the bg class today is not my day 😅

**Pull request merged**
